gen/api_defs.jl

```diff
@@ -273,7 +287,8 @@
 ###
 
 @bind h5do_append(dset_id::hid_t, dxpl_id::hid_t, index::Cuint, num_elem::hsize_t, memtype::hid_t, buffer::Ptr{Cvoid})::herr_t "error appending"
-@bind h5do_write_chunk(dset_id::hid_t, dxpl_id::hid_t, filter_mask::Int32, offset::Ptr{hsize_t}, bufsize::Csize_t, buf::Ptr{Cvoid})::herr_t "Error writing chunk"
+# h5do_write_chunk is deprecated as of hdflib 1.10.3
+@bind h5do_write_chunk(dset_id::hid_t, dxpl_id::hid_t, filter_mask::UInt32, offset::Ptr{hsize_t}, bufsize::Csize_t, buf::Ptr{Cvoid})::herr_t "Error writing chunk"
```

Comment:
I think we might be able to just delete this since the minimum hdf5lib version we support is 1.10.4

Reply:
`h5do_write_chunk` still exists in 1.10.4, it's just deprecated. Perhaps we should mark it as deprecated in case anyone is using it?

Reply:
Ideally, we could mark it as deprecated, but the auto-bind generator doesn't support that.  I guess we can just leave it as is.